src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -6088,39 +6088,43 @@ std::optional<ProjectionCandidate> MergeTreeData::getQueryProcessingStageWithAgg
     if (query_info.additional_filter_ast)
         return std::nullopt;
 
-    auto query_ptr = query_info.original_query;
+    auto query_ptr = query_info.query;
+    auto original_query_ptr = query_info.original_query;
```

Comment:
This makes the code convoluted, can we avoid it?

Reply:
Avoid exactly what? We need to access both original and modified query, check: https://github.com/ClickHouse/ClickHouse/pull/40945#discussion_r964782102. I could alway de-reference `query_info` instead, but I don't think it's any better